portal/static/js/admin.js

```diff
@@ -261,7 +263,47 @@
                     if (match.length === 0) {
                         return false;
                     }
-                    $("#patientListOptions .or, #createUserLink").addClass("disabled").css("display", "none");
+                    self.setCreateAccountVis(true);
+                    self.checkAdmin();
+                });
+            },
+            setCreateAccountVis: function(hide) {
+                var createAccountElements = $("#patientListOptions .or, #createUserLink");
+                if (hide) {
+                    createAccountElements.css("display", "none");
+                    return;
+                }
+                createAccountElements.css("display", "block");
+            },
+            getUserRoles: function(callback) {
+                callback = callback || function() {};
+                if (this.userRoles.length > 0) {
+                    callback(this.userRoles);
+                    return;
+                }
+                this.setUserRoles(callback);
+            },
+            setUserRoles: function(callback) {
+                callback = callback || function() {};
+                var self = this, tnthAjax = this.getDependency("tnthAjax");
+                tnthAjax.getRoles(this.userId, function(data) {
```

Comment:
Naive question, but couldn't you just check the returned list from `tnthAjax.getRoles()` instead of assigning them to `userRoles` and checking that?